### PR TITLE
shouldIntersect 

### PR DESCRIPTION
**Pull request by AlexCue987**
`shouldIntersect` verifies if two Sets intersect

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.kotest.assertions.print.print
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+
+infix fun <T> Set<T>.shouldIntersect(ts: Set<T>) = this should intersectWith(ts)
+infix fun <T> Set<T>.shouldNotIntersect(ts: Set<T>) = this shouldNot intersectWith(ts)
+
+fun <T> intersectWith(ts: Set<T>): Matcher<Set<T>> = object : Matcher<Set<T>> {
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt]: 🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

File: kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   override fun test(value: Set<T>): MatcherResult {
+      val intersection = ts.filter { value.contains(it) }
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt]: https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/intersect.html

Can't actually believe this function exists in stdlib

```suggestion
      val intersection = ts intercect value
```

**Review by LeoColman — CHANGES_REQUESTED** [review]

File: kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt (hunk 1 above)
Comment (AlexCue987) [reply kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt]: > 🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

will do thx

Comment (AlexCue987) [reply kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt]: > 🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

done

**Review by LeoColman — APPROVED** [review]

**Pull request merged**